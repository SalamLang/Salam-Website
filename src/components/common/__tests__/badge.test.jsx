import React from 'react';
import {Badge} from '../Badge';

// Mock the class-variance-authority to test component structure
const mockSpan = (config) => {
  return (props) => {
    // Simulate the cva behavior
    const baseClass = "span";
    const intentClasses = {
      primary: "bg-light-primary text-primary",
      success: "bg-light-success text-success", 
      warning: "bg-light-warning text-warning",
      danger: "bg-light-danger text-danger"
    };
    const sizeClasses = {
      medium: "py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs"
    };
    const roundedClasses = {
      full: "rounded-full"
    };
    
    const intent = props.intent || 'primary';
    const size = props.size || 'medium';
    const rounded = props.rounded || 'full';
    const className = props.className || '';
    
    return [baseClass, intentClasses[intent], sizeClasses[size], roundedClasses[rounded], className].filter(Boolean).join(' ');
  };
};

describe('Badge component', () => {
  it('should be a function', () => {
    expect(typeof Badge).toBe('function');
  });

  it('should render with correct props structure', () => {
    const props = { className: "primary", children: "آنلاین کار کن" };
    const element = Badge(props);
    
    // Check that it returns a React element
    expect(element).toBeTruthy();
    expect(element.type).toBe('span');
  });

  it('should handle different intent props', () => {
    const successBadge = Badge({ intent: "success", children: "کوتاه بودن" });
    const warningBadge = Badge({ intent: "warning", children: "زیبا بودن" });
    const dangerBadge = Badge({ intent: "danger", children: "فارسی بودن" });
    
    expect(successBadge.type).toBe('span');
    expect(warningBadge.type).toBe('span');
    expect(dangerBadge.type).toBe('span');
  });

  it('should accept additional props', () => {
    const badge = Badge({ 
      intent: "primary", 
      className: "extra-class",
      'data-testid': 'test-badge',
      children: "Test"
    });
    
    expect(badge.type).toBe('span');
    expect(badge.props['data-testid']).toBe('test-badge');
  });

  it('component structure validation', () => {
    // Test that the component can be created without errors
    const testCases = [
      { className: "primary", children: "آنلاین کار کن" },
      { className: "primary medium full", children: "آنلاین کار کن" },
      { intent: "success", children: "کوتاه بودن" },
      { intent: "warning", children: "زیبا بودن" },
      { intent: "danger", children: "فارسی بودن" }
    ];
    
    testCases.forEach((props, index) => {
      const badge = Badge(props);
      expect(badge).toBeTruthy();
      expect(badge.type).toBe('span');
      expect(badge.props.children).toBe(props.children);
    });
  });
});
