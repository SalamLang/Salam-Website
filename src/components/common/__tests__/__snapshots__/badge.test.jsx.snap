// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`Badge component renders with primary className correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "آنلاین کار کن",
  ],
  "props": {
    "className": "span bg-light-primary text-primary py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs rounded-full primary",
  },
  "type": "span",
}
`;

exports[`Badge component renders with multiple classNames correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "آنلاین کار کن",
  ],
  "props": {
    "className": "span bg-light-primary text-primary py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs rounded-full primary medium full",
  },
  "type": "span",
}
`;

exports[`Badge component renders with success intent correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "کوتاه بودن",
  ],
  "props": {
    "className": "span bg-light-success text-success py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs rounded-full",
  },
  "type": "span",
}
`;

exports[`Badge component renders with warning intent correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "زیبا بودن",
  ],
  "props": {
    "className": "span bg-light-warning text-warning py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs rounded-full",
  },
  "type": "span",
}
`;

exports[`Badge component renders with danger intent correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "فارسی بودن",
  ],
  "props": {
    "className": "span bg-light-danger text-danger py-1 sm:py-2 px-2 sm:px-4 font-Estedad-Medium text-[10px] sm:text-xs rounded-full",
  },
  "type": "span",
}
`;
